jenkins/databricks/run-tests.py

```diff
@@ -35,7 +35,7 @@ def main():
   print("rsync command: %s" % rsync_command)
   subprocess.check_call(rsync_command, shell = True)
 
-  ssh_command = "bash -c 'ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null ubuntu@%s -p 2200 -i %s %s %s %s 2>&1 | tee testout; if [ `echo ${PIPESTATUS[0]}` -ne 0 ]; then false; else true; fi'" % (master_addr, params.private_key_file, params.script_dest, params.jar_path, params.spark_conf)
+  ssh_command = "bash -c 'ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null ubuntu@%s -p 2200 -i %s %s %s %s 2>&1 | tee testout; if [ `echo ${PIPESTATUS[0]}` -ne 0 ]; then false; else true; fi'" % (master_addr, params.private_key_file, params.script_dest, params.jar_path, params.spark_conf, params.base_spark_pom_version)
```

Comment:
Need to add a input placeholder here :  

 `-i %s %s %s %s 2>&1`  to 

`-i %s %s %s %s %s 2>&1`

Reply:
Is there a way we can test the pipeline before I merge this in? I have tested test.sh on databricks but I would like to test it via Jenkins before its merged if possible

Reply:
Sure, I'll run the Jenkins pipeline before it get merged.

Reply:
I'll tested with Jenkins pipeline, there is a problem here with the params parsing logic: https://github.com/NVIDIA/spark-rapids/blob/branch-21.08/jenkins/databricks/run-tests.py#L38

In `test only` pipeline, we must set `[params.jar_path(LOCAL_JAR_PATH=$1), params.spark_conf(SPARK_CONF=$2](https://github.com/NVIDIA/spark-rapids/blob/branch-21.08/jenkins/databricks/test.sh#L20-L21))`  to point the where the downloaded jar path is, and set some extral configs to run tests.

CLI: 

bash -c 'ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null ubuntu@ec2-54-187-65-221.us-west-2.compute.amazonaws.com -p 2200 -i **** /home/ubuntu/test.sh   ** /home/ubuntu/jars  spark.rapids.cudfVersionOverride=true 3.1.1 ** 2>&1 | tee testout; if [ `echo ${PIPESTATUS[0]}` -ne 0 ]; then false; else true; fi'

```
LOCAL_JAR_PATH=/home/ubuntu/jars
SPARK_CONF=spark.rapids.cudfVersionOverride=true
BASE_SPARK_VER=3.1.1
```


But in the `build + test` pipeline, we must unset these params, as the jars are built out in diff subdirs e.g.  `spark-rapids/dist/target/rapids-4-spark*.jar. spark-rapids/udf-examples/target/*.jars`

CLI : 

bash -c 'ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null ubuntu@ec2-54-187-65-221.us-west-2.compute.amazonaws.com -p 2200 -i **** /home/ubuntu/test.sh   **3.1.1** 2>&1 | tee testout; if [ `echo ${PIPESTATUS[0]}` -ne 0 ]; then false; else true; fi'

```
LOCAL_JAR_PATH=3.1.1
SPARK_CONF=
BASE_SPARK_VER=
```

Then  `build + test` pipeline will failed due to paraems $1 $2 are missing/unset, BASE_SPARK_VER because $1 instead of $3

To fix this issue, we can change to explicitly set each params : `LOCAL_JAR_PATH=%s SPARK_CONF=%s BASE_SPARK_VER=%s bash test.sh` as below in the run-test.py, and easier to extend params later.

https://github.com/NVIDIA/spark-rapids/blob/branch-21.08/jenkins/databricks/run-tests.py#L38  -->

`ssh_command = "ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null ubuntu@%s -p 2200 -i %s 'LOCAL_JAR_PATH=%s SPARK_CONF=%s BASE_SPARK_VER=%s bash %s 2>&1 | tee testout; if [ `echo ${PIPESTATUS[0]}` -ne 0 ]; then false; else true; fi'" % (master_addr, params.private_key_file, params.jar_path, params.spark_conf, params.base_spark_pom_version, params.script_dest)`

Reply:
@razajafri  @tgravescs  Could you please help to review:

Reply:
Cache test PASS with Jenkins pipeline: 2:40:52  To adjust logging level use sc.setLogLevel(newLevel). For SparkR, use setLogLevel(newLevel).
12:40:52  
12:40:52  
12:40:52  [32m======================= [32m[1m975 passed[0m[32m in 733.11s (0:12:13)[0m[32m ========================[0m
12:40:52  bash: line 0: [: too many arguments